lib/src/widgets/post/post_widget.dart

```diff
@@ -386,90 +414,23 @@ class _NovaPostWidgetState extends State<NovaPostWidget> {
-                              },
-                              border: const Border(),
-                              title: LMTextView(
-                                text: linkAttachment
-                                        ?.attachmentMeta.ogTags?.title ??
-                                    "--",
-                                maxLines: 1,
-                                overflow: TextOverflow.ellipsis,
-                                textStyle: theme.textTheme.titleMedium,
-                              ),
-                              subtitle: LMTextView(
-                                text: linkAttachment
-                                        ?.attachmentMeta.ogTags?.description ??
-                                    "--",
-                                maxLines: 2,
-                                overflow: TextOverflow.ellipsis,
-                                textStyle: theme.textTheme.displayMedium,
-                              ),
-                            )
-                          : SizedBox(
-                              child: GestureDetector(
-                                behavior: HitTestBehavior.opaque,
-                                onTap: () async {
-                                  await videoController?.player.pause();
-                                  await Navigator.push(
-                                    context,
-                                    MaterialPageRoute(
-                                      builder: (context) {
-                                        return MediaPreview(
-                                          postAttachments:
-                                              postDetails!.attachments!,
-                                          post: postDetails!.toPost(),
-                                          user: widget.user,
-                                        );
-                                      },
-                                    ),
-                                  );
-                                  await videoController?.player.play();
-                                },
-                                child: LMPostMedia(
-                                  initialiseVideoController:
-                                      (VideoController controller) {
-                                    videoController = controller;
-                                  },
-                                  attachments: postDetails!.attachments!,
-                                  borderRadius: 16.0,
-                                  height: screenSize.width - 32,
-                                  width: screenSize.width - 32,
-                                  boxFit: BoxFit.cover,
-                                  textColor: ColorTheme
-                                      .novaTheme.colorScheme.onPrimary,
+                      ? postDetails!.isRepost
+                          ? postFromRepost(postDetails!, widget.repostedPost) !=
```

Comment:
if the post is a repost, then the data must not be null. Why do we need thiw check here